### ISSUE TITLE
How can I iterate through Menu from API and pass in a param

### ISSUE THREAD

**Issue by armagedan** [react-navigation/react-navigation#6166]
So this question maybe a bit broad so I will split this in to two.

**Part 1:**

Basically I want to iterate through my menu which is coming from a server API. This has a lot of menu items and here is my `Router.js`

    import React, { Component } from "react";
    import { createSwitchNavigator, createStackNavigator, createAppContainer, createDrawerNavigator } from 'react-navigation';

    import Home from "./src/Screens/Home";
    import SinglePage from "./src/Screens/SinglePage";
    import CatPage from './src/Screens/CatPage';

    const AppStackNavigator = createStackNavigator({

    Home: { screen: Home },
    SinglePage: { screen: SinglePage },
    CatPage: { screen: CatPage, params: { TeSt: '123' } },

    },

    {
        initialRouteName: 'Home',
        mode: "card",
        headerMode: "none",
        //cardStyle: { backgroundColor: '#757575' },
    }
    );

    const Project = createDrawerNavigator({

    Home: AppStackNavigator,

    Something: AppStackNavigator,

    },

    {
        hideStatusBar: false,
        initialRouteName: 'Home',
        drawerPosition: 'right',
        drawerBackgroundColor: 'rgba(255,255,255,.9)',
        overlayColor: '#6b52ae',
        contentOptions: {
            activeTintColor: '#fff',
            activeBackgroundColor: '#6b52ae',
        },
    }

    );
 

    export default createAppContainer(Project);`

So this is my router file which is in the root directory. What I want is to use this:

    async getMenu(){

        axios.get('http://192.168.1.5/api/menus')
            .then(response => {
                this.setState({menus: response.data.menus});
            })
            .catch(erorr => {
                console.log(erorr)
            })

      }`

inside the router file and iterate the menu items with its ID as the param.

**Part 2:**

Part two is something I can't wrap my head around. Basically After iteration, I want each menu item to be as an Item inside the drawer menu. And I want each of them to pass a parameter (in this case the id of the menu item) to a single stack navigator. Basically I will have one single stack navigator which will accept params from drawer navigator screen/item. Suppose `Shop` inside the drawer navigator will pass a param to `CatPage` inside of stack navigator. How can I achieve this?

In a nutshell, what I am trying to do here is this:

    const Project = createDrawerNavigator({

        Home: AppStackNavigator,
    
        Something: {screen: CatSingle, params: {test:123}},
        Something1: {screen: CatSingle, params: {test:123}},
        Something2: {screen: CatSingle, params: {test:123}},
        Something3: {screen: CatSingle, params: {test:123}},
        Something4: {screen: CatSingle, params: {test:123}},

    },

Iterate using Axios by calling a an API. As of the above example works. But how can I iterate through my menu inside of this

**Comment by brentvatne** [comment]
hi there! this is a question which is better suited for stackoverflow or discord. see https://reactnavigation.org/help.html for more information on how to get help.
